### PR TITLE
Refactored fsck to more easily support new provenance model

### PR DESCRIPTION
**Pull request by zmajeed**

### CODE REVIEW

File: src/server/pfs/server/driver_fsck.go [hunk 1]
```diff
@@ -134,163 +134,231 @@ func (e ErrMissingBranchHead) Error() string {
...
-	if err := d.repos.ReadOnly(ctx).List(repoInfo, col.DefaultOptions(), func(string) error {
-		commitInfo := &pfs.CommitInfo{}
-		if err := d.commits.ReadOnly(ctx).GetByIndex(pfsdb.CommitsRepoIndex, pfsdb.RepoKey(repoInfo.Repo), commitInfo, col.DefaultOptions(), func(string) error {
-			commitInfos[pfsdb.CommitKey(commitInfo.Commit)] = proto.Clone(commitInfo).(*pfs.CommitInfo)
-			return nil
+	return nil
+}
+
+func checkBranchSubvenances(bi *pfs.BranchInfo, branchInfos map[string]*pfs.BranchInfo, onError func(error) error) error {
+
```
Comment (acohen4) [inline src/server/pfs/server/driver_fsck.go:164]: nit: could you remove the in function new lines like here?

File: src/server/pfs/server/driver_fsck.go [hunk 2]
```diff
@@ -134,163 +134,231 @@ func (e ErrMissingBranchHead) Error() string {
...
-		if !equalBranches(append(bi.Provenance, bi.Branch), union) {
-			if err := onError(ErrBranchProvenanceTransitivity{
-				BranchInfo:     bi,
-				FullProvenance: union,
+	}
+
+	return nil
+}
+
+func CheckParentCommit(commitInfo *pfs.CommitInfo, commitInfos map[string]*pfs.CommitInfo, onError func(error) error) error {
```
Comment (acohen4) [inline src/server/pfs/server/driver_fsck.go:242]: nit: could you camel case these internal functions? upper-casing them exposes them as "public" in the go package which is not clear that we want right now

**Review by acohen4 — APPROVED** [review]
a few nits but overall this LGTM!

**Comment by zmajeed** [comment]
closed in favor of https://github.com/pachyderm/pachyderm/pull/8715

**Pull request closed without merging**